### PR TITLE
#3 refactor composite actions

### LINKED ISSUE #3: Refactor composite actions output
**Overview of the issue**

The _infrastructure as code_ Terraform composite action is outputing important values as an artifact, called `terraform`, containing a JSON.

However, in able to support bicep deployment as well, we need to move from that mode to something more universal. It'll also allow people to reuse these composite actions outside of the use of NubesGen.

**TODO**
- [ ] Generate outputs for Bicep deployment - See https://github.com/cmaneu/proxyvote/blob/0efc04dfa952221b5f831eab4f2fdffc0d73f24f/.github/workflows/deploy-app.yaml#L55
- [ ] In main repo, update Terraform template to match output names (not `application_caf_name` but `application_name`
- [ ] In main repo, update the github action mustache template. Let the `manage-infrastructure` step declare proper outputs
```yaml
  manage-infrastructure:
    runs-on: ubuntu-20.04
    outputs: 
      application_name: ${{ steps.apply-terraform.outputs.application_name }}
    steps:
      - name: Apply Terraform configuration
        id: apply-terraform
        uses: microsoft/nubesgen-actions/gitops-apply-terraform@3-refactor-composite-actions
        with:
          azure_credentials: ${{ secrets.AZURE_CREDENTIALS }}
          tf_storage_account: ${{ secrets.TF_STORAGE_ACCOUNT }}
  deploy:
    needs: [manage-infrastructure, build]
    runs-on: ubuntu-20.04
    steps:
      - name: Get outputs
        env: 
          APPNAME: ${{ needs.manage-infrastructure.outputs.application_name }}
        run: echo '$APPNAME'
        shell: bash
      - run: echo '${{ toJson(needs) }}'
        shell: bash        
      - name: Deploy a .NET application to Azure App Service
        uses: microsoft/nubesgen-actions/gitops-deploy-to-app-service-dotnet@3-refactor-composite-actions
        with:
          azure_credentials: ${{ secrets.AZURE_CREDENTIALS }}
          application_name: ${{ needs.manage-infrastructure.outputs.application_name }}
```
- [ ] Update all deployment composite actions
- [ ] Remove debug toJson steps

### PR DESCRIPTION
**Pull request by cmaneu**
Fixes #3

### CODE REVIEW

File: gitops-apply-bicep/action.yml [hunk 1]
```diff
@@ -10,6 +10,22 @@ inputs:
...
     required: true  
+outputs: 
+  application_name:
+    description: 'Name of the application'
+    value:     
+  application_url: 
+    description: 'URL of the application'
+    value:     
+  resource_group: 
+    description: 'Resource group name where the resources have been deployed'
```
Comment (jdubois) [inline gitops-apply-bicep/action.yml:21]:
```suggestion
    description: 'Resource group name where the resources has been deployed'
```

File: gitops-apply-bicep/action.yml [hunk 2]
```diff
@@ -10,6 +10,22 @@ inputs:
...
+    value:     
+  container_registry_name: 
+    description: 'The Container registry name that have been deployed'
```
Comment (jdubois) [inline gitops-apply-bicep/action.yml:24]:
```suggestion
    description: 'The Container registry name that has been deployed'
```

File: gitops-apply-terraform/action.yml [hunk 3]
```diff
@@ -11,6 +11,22 @@ inputs:
...
     default: 'rg-terraform-001'
+outputs: 
+  application_name:
+    description: 'Name of the application'
+    value: ${{ steps.terraform-output.outputs.application_name }}    
+  application_url: 
+    description: 'URL of the application'
+    value: ${{ steps.terraform-output.outputs.application_url }}    
+  resource_group: 
+    description: 'Resource group name where the resources have been deployed'
```
Comment (jdubois) [inline gitops-apply-terraform/action.yml:22]:
```suggestion
    description: 'Resource group name where the resources has been deployed'
```

File: gitops-apply-terraform/action.yml [hunk 4]
```diff
@@ -11,6 +11,22 @@ inputs:
...
+    value: ${{ steps.terraform-output.outputs.resource_group }}    
+  container_registry_name: 
+    description: 'The Container registry name that have been deployed'
```
Comment (jdubois) [inline gitops-apply-terraform/action.yml:25]:
```suggestion
    description: 'The Container registry name that have has deployed'
```

File: gitops-build-and-deploy-to-app-service-docker-quarkus-native/action.yml [hunk 5]
```diff
@@ -4,18 +4,15 @@ inputs:
   azure_credentials:
     description: 'Azure Credentials, stored in a secret on your GitHub repository, generated by https://github.com/microsoft/NubesGen/blob/main/docs/gitops-quick-start.md'
     required: true
+  application_name: 
+    description: 'The name of the application where to deploy'
```
Comment (jdubois) [inline gitops-build-and-deploy-to-app-service-docker-quarkus-native/action.yml:8]:
```suggestion
    description: 'The name of the application that will be deployed'
```

File: gitops-build-and-deploy-to-app-service-docker-spring/action.yml [hunk 6]
```diff
@@ -4,18 +4,16 @@ inputs:
   azure_credentials:
     description: 'Azure Credentials, stored in a secret on your GitHub repository, generated by https://github.com/microsoft/NubesGen/blob/main/docs/gitops-quick-start.md'
     required: true
+  application_name: 
+    description: 'The name of the application where to deploy'
```
Comment (jdubois) [inline gitops-build-and-deploy-to-app-service-docker-spring/action.yml:8]:
```suggestion
    description: 'The name of the application that will be deployed'
```

File: gitops-build-and-deploy-to-app-service-docker/action.yml [hunk 7]
```diff
@@ -4,20 +4,18 @@ inputs:
   azure_credentials:
     description: 'Azure Credentials, stored in a secret on your GitHub repository, generated by https://github.com/microsoft/NubesGen/blob/main/docs/gitops-quick-start.md'
     required: true
+  application_name: 
+    description: 'The name of the application where to deploy'
```
Comment (jdubois) [inline gitops-build-and-deploy-to-app-service-docker/action.yml:8]:
```suggestion
    description: 'The name of the application that will be deployed'
```

File: gitops-build-function-java-gradle/action.yml [hunk 8]
```diff
@@ -9,18 +9,15 @@ inputs:
     description: 'The command to build the Java project'
     required: false
     default: 'gradle build azureFunctionsPackage -Pprod,azure'
+  application_name: 
+    description: 'The name of the application where to deploy'
```
Comment (jdubois) [inline gitops-build-function-java-gradle/action.yml:13]:
```suggestion
    description: 'The name of the application that will be deployed'
```

File: gitops-build-function-java-maven/action.yml [hunk 9]
```diff
@@ -9,18 +9,15 @@ inputs:
     description: 'The command to build the Java project'
     required: false
     default: 'mvn package azure-functions:package -Pprod,azure'
+  application_name: 
+    description: 'The name of the application where to deploy'
```
Comment (jdubois) [inline gitops-build-function-java-maven/action.yml:13]:
```suggestion
    description: 'The name of the application that will be deployed'
```

File: gitops-deploy-to-app-service-dotnet/action.yml [hunk 10]
```diff
@@ -4,21 +4,20 @@ inputs:
   azure_credentials:
     description: 'Azure Credentials, stored in a secret on your GitHub repository, generated by https://github.com/microsoft/NubesGen/blob/main/docs/gitops-quick-start.md'
     required: true
+  application_name: 
+    description: 'The name of the application where to deploy'
```
Comment (jdubois) [inline gitops-deploy-to-app-service-dotnet/action.yml:8]:
```suggestion
    description: 'The name of the application that will be deployed'
```

File: gitops-deploy-to-app-service-dotnet/action.yml [hunk 11]
```diff
@@ -4,21 +4,20 @@ inputs:
...
+    required: true
+  artifact_name:
+    description: 'The name of the artifact containing the app to deploy.'
```
Comment (jdubois) [inline gitops-deploy-to-app-service-dotnet/action.yml:11]:
```suggestion
    description: 'The name of the artifact containing the application to deploy.'
```

File: gitops-deploy-to-app-service-java/action.yml [hunk 12]
```diff
@@ -4,27 +4,26 @@ inputs:
   azure_credentials:
     description: 'Azure Credentials, stored in a secret on your GitHub repository, generated by https://github.com/microsoft/NubesGen/blob/main/docs/gitops-quick-start.md'
     required: true
+  application_name: 
+    description: 'The name of the application where to deploy'
```
Comment (jdubois) [inline gitops-deploy-to-app-service-java/action.yml:8]:
```suggestion
    description: 'The name of the application that will be deployed'
```

File: gitops-deploy-to-app-service-java/action.yml [hunk 13]
```diff
@@ -4,27 +4,26 @@ inputs:
...
+    required: true
+  artifact_name:
+    description: 'The name of the artifact containing the app to deploy.'
```
Comment (jdubois) [inline gitops-deploy-to-app-service-java/action.yml:11]:
```suggestion
    description: 'The name of the artifact containing the application to deploy.'
```

File: gitops-deploy-to-app-service-nodejs/action.yml [hunk 14]
```diff
@@ -4,21 +4,20 @@ inputs:
   azure_credentials:
     description: 'Azure Credentials, stored in a secret on your GitHub repository, generated by https://github.com/microsoft/NubesGen/blob/main/docs/gitops-quick-start.md'
     required: true
+  application_name: 
+    description: 'The name of the application where to deploy'
```
Comment (jdubois) [inline gitops-deploy-to-app-service-nodejs/action.yml:8]:
```suggestion
    description: 'The name of the application that will be deployed'
```

File: gitops-deploy-to-app-service-nodejs/action.yml [hunk 15]
```diff
@@ -4,21 +4,20 @@ inputs:
...
+    required: true
+  artifact_name:
+    description: 'The name of the artifact containing the app to deploy.'
```
Comment (jdubois) [inline gitops-deploy-to-app-service-nodejs/action.yml:11]:
```suggestion
    description: 'The name of the artifact containing the application to deploy.'
```

File: gitops-deploy-to-function-dotnet/action.yml [hunk 16]
```diff
@@ -4,21 +4,20 @@ inputs:
   azure_credentials:
     description: 'Azure Credentials, stored in a secret on your GitHub repository, generated by https://github.com/microsoft/NubesGen/blob/main/docs/gitops-quick-start.md'
     required: true
+  application_name: 
+    description: 'The name of the application where to deploy'
```
Comment (jdubois) [inline gitops-deploy-to-function-dotnet/action.yml:8]:
```suggestion
    description: 'The name of the application that will be deployed'
```

File: gitops-deploy-to-function-dotnet/action.yml [hunk 17]
```diff
@@ -4,21 +4,20 @@ inputs:
...
+    required: true
+  artifact_name:
+    description: 'The name of the artifact containing the app to deploy.'
```
Comment (jdubois) [inline gitops-deploy-to-function-dotnet/action.yml:11]:
```suggestion
    description: 'The name of the artifact containing the application to deploy.'
```

File: gitops-deploy-to-function-java/action.yml [hunk 18]
```diff
@@ -4,28 +4,26 @@ inputs:
   azure_credentials:
     description: 'Azure Credentials, stored in a secret on your GitHub repository, generated by https://github.com/microsoft/NubesGen/blob/main/docs/gitops-quick-start.md'
     required: true
+  application_name: 
+    description: 'The name of the application where to deploy'
```
Comment (jdubois) [inline gitops-deploy-to-function-java/action.yml:8]:
```suggestion
    description: 'The name of the application that will be deployed'
```

File: gitops-deploy-to-function-java/action.yml [hunk 19]
```diff
@@ -4,28 +4,26 @@ inputs:
...
+    required: true
+  artifact_name:
+    description: 'The name of the artifact containing the app to deploy.'
```
Comment (jdubois) [inline gitops-deploy-to-function-java/action.yml:11]:
```suggestion
    description: 'The name of the artifact containing the application to deploy.'
```

File: gitops-deploy-to-function-nodejs/action.yml [hunk 20]
```diff
@@ -4,21 +4,20 @@ inputs:
   azure_credentials:
     description: 'Azure Credentials, stored in a secret on your GitHub repository, generated by https://github.com/microsoft/NubesGen/blob/main/docs/gitops-quick-start.md'
     required: true
+  application_name: 
+    description: 'The name of the application where to deploy'
```
Comment (jdubois) [inline gitops-deploy-to-function-nodejs/action.yml:8]:
```suggestion
    description: 'The name of the application that will be deployed'
```

File: gitops-deploy-to-function-nodejs/action.yml [hunk 21]
```diff
@@ -4,21 +4,20 @@ inputs:
...
+    required: true
+  artifact_name:
+    description: 'The name of the artifact containing the app to deploy.'
```
Comment (jdubois) [inline gitops-deploy-to-function-nodejs/action.yml:11]:
```suggestion
    description: 'The name of the artifact containing the application to deploy.'
```

File: gitops-deploy-to-spring-cloud-java/action.yml [hunk 22]
```diff
@@ -8,19 +8,19 @@ inputs:
     description: 'Use staging deployement (blue/green deployment)'
     required: false
     default: false
+  application_name: 
+    description: 'The name of the application where to deploy'
```
Comment (jdubois) [inline gitops-deploy-to-spring-cloud-java/action.yml:12]:
```suggestion
    description: 'The name of the application that will be deployed'
```

File: gitops-deploy-to-spring-cloud-java/action.yml [hunk 23]
```diff
@@ -8,19 +8,19 @@ inputs:
...
+    required: true
+  artifact_name:
+    description: 'The name of the artifact containing the app to deploy.'
```
Comment (jdubois) [inline gitops-deploy-to-spring-cloud-java/action.yml:15]:
```suggestion
    description: 'The name of the artifact containing the application to deploy.'
```

**Comment by jdubois** [comment]
@cmaneu I'm starting to review this, is there any reason why you kept it as a draft?

**Comment by cmaneu** [comment]
Yes. This branch points to a feature branch in the nubesgsn-action repo. Before moving it to a PR status, we should merge the PR on the nubesgen-action before.

**Pull request merged**